stacker/util.py

```diff
@@ -448,3 +452,168 @@ def ensure_s3_bucket(s3_client, bucket_name, bucket_region):
             logger.exception("Error creating bucket %s. Error %s",
                              bucket_name, e.response)
             raise
+
+
+class SourceProcessor():
+    """Makes remote python package sources available in the running python
+       environment."""
+
+    def __init__(self, stacker_cache_dir=None):
+        """
+        Processes a config's list of package sources
+
+        Args:
+            stacker_cache_dir (string): Directory of stacker local cache.
+                                        Default to $HOME/.stacker
+        """
+        stacker_cache_dir = (stacker_cache_dir or
+                             os.path.join(os.environ['HOME'], '.stacker'))
+        package_cache_dir = os.path.join(stacker_cache_dir, 'packages')
+        self.stacker_cache_dir = stacker_cache_dir
+        self.package_cache_dir = package_cache_dir
+        self.create_cache_directories()
+
+    def create_cache_directories(self):
+        """Ensures that SourceProcessor cache directories exist.
+
+           Returns True to indicate the cache directories are present."""
+        if not os.path.isdir(self.package_cache_dir):
+            if not os.path.isdir(self.stacker_cache_dir):
+                os.mkdir(self.stacker_cache_dir)
+            os.mkdir(self.package_cache_dir)
+        return True
+
+    def get_package_sources(self, sources):
+        """Makes remote python packages available for local use
+
+        Args:
+            sources (dict): Dictionary of remote sources from config.
+                            Currently supports git repositories
+            Example:
+              {'git': [
+                  {'uri': 'git@github.com:remind101/stacker_blueprints.git',
+                   'tag': '1.0.0',
+                   'paths': ['stacker_blueprints']},
+                  {'uri': 'git@github.com:acmecorp/stacker_blueprints.git'},
+                  {'uri': 'git@github.com:contoso/webapp.git',
+                   'branch': 'staging'},
+                  {'uri': 'git@github.com:contoso/foo.git',
+                   'commit': '12345678'}
+              ]}
+
+        """
+        # Checkout git repositories specified in config
+        if 'git' in sources:
+            for config in sources['git']:
+                self.fetch_git_package(config=config)
+
+    def fetch_git_package(self, config):
+        """Makes a remote git repository available for local use
+
+        Args:
+            config (dict): Dictionary of git repo configuration
+
+        """
+        # If a specific commit or tag is defined, first check to see if it is
+        # already cached.
+        if config.get('commit'):
```

Comment:
Worth turning this whole block into it's own function/method and writing some tests around it?

Also, do you think it's worth catching if someone specifies both a commit and tag?